src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -80,4 +74,60 @@ public GHClient(IProgram program)
         {
         }
     }
+
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
+    [ProvideService(typeof(IUIProvider), IsAsyncQueryable = true)]
+    [ProvideAutoLoad(UIContextGuids.NoSolution)]
+    [ProvideAutoLoad(UIContextGuids.SolutionExists)]
+    [Guid("D5CE1488-DEDE-426D-9E5B-BFCCFBE33E53")]
+    public sealed class ServiceProviderPackage : AsyncPackage
+    {
+        static Version vsversion;
+        public static Version VSVersion
+        {
+            get
+            {
+                if (vsversion == null)
```

Comment:
`InitializeAsync` only runs once (guaranteed), it's just a way of loading packages in a background thread, so reentrancy shouldn't be a problem right now, given that this property is only used for the package check. It might be an issue if other code uses it though, so maybe it shouldn't be static nor public.